### ISSUE TITLE
[Bug]: Error while downloading

### ISSUE THREAD

**Issue by thedtech** [Puyodead1/udemy-downloader#222]
### What happened?

Script is throwing error in the beginning after extracting info dialogues. This error occurs with this particular course in my case (Can be with other courses also) : [https://www.udemy.com/course/react-the-complete-guide-incl-redux/](url)

This course doesn't have any DRM.

### Expected Result

It should download normally like other courses.

### Branch

master/main

### What operating systems are you seeing the problem on?

Windows

### Relevant log output

```shell
C:\uDL>main.py -c https://www.udemy.com/course/react-the-complete-guide-incl-redux/ --download-assets --download-captions --download-quizzes --id-as-course-name --log-level ERROR
Traceback (most recent call last):
  File "C:\Python3\Lib\site-packages\requests\models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python3\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python3\Lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python3\Lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\uDL\main.py", line 2068, in <module>
    main()
  File "C:\uDL\main.py", line 1951, in main
    course_json = udemy._extract_course_json(course_url, course_id, portal_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\uDL\main.py", line 770, in _extract_course_json
    resp = self._extract_large_course_content(url=url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\uDL\main.py", line 788, in _extract_large_course_content
    resp = self.session._get(_next).json()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python3\Lib\site-packages\requests\models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

### Other information

![Screenshot (4)](https://github.com/Puyodead1/udemy-downloader/assets/28692876/8de5c43c-6897-4f79-a4a4-6cf48bc85fa0)
![Screenshot (5)](https://github.com/Puyodead1/udemy-downloader/assets/28692876/1fd1c92b-6b40-4071-afe6-75f0d7105098)

**Comment by thedtech** [comment]
> are you passing a bearer token? and use `https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn`

Yes, I am using the token.
Even after using [https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn](url), Error is same. If you want, I can provide credentials for test.

![Screenshot (6)](https://github.com/Puyodead1/udemy-downloader/assets/28692876/50b86209-8b99-4996-9b4b-14838db16481)

**Comment by HAP5dew** [comment]
@thedtech can you help me with coding please or just enroll Udemy courses. im trying to search for centuries for free paid courses but can find anything and don't know anything with code. Can we contact?

**Comment by mcs-nuuk** [comment]
I've got the same issue unfortunately.

**Comment by Puyodead1** [comment]
> > are you passing a bearer token? and use `https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn`
> 
> Yes, I am using the token. Even after using [https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn](url), Error is same. If you want, I can provide credentials for test.
> 
> ![Screenshot (6)](https://private-user-images.githubusercontent.com/28692876/337727575-50b86209-8b99-4996-9b4b-14838db16481.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3OTI0OTYsIm5iZiI6MTcxODc5MjE5NiwicGF0aCI6Ii8yODY5Mjg3Ni8zMzc3Mjc1NzUtNTBiODYyMDktOGI5OS00OTk2LTliNGItMTQ4MzhkYjE2NDgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDEwMTYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjODI4MTA2NDVmNzVmNjE1MDIxYjg1Njk0MTkzMDdmOTQwYzM5NzRmOTcyM2Q2ZmI0MjNhN2M4NWJlOTZiZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.d3FRoTiwiQ_pmYE3zzSjGnq5DarhPkknsN9TkIrxmF4)

I can look into it, but yes i will need at least a bearer token to test with. you can email me puyodead@proton.me

**Comment by thedtech** [comment]
> > > are you passing a bearer token? and use `https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn`
> > 
> > Yes, I am using the token. Even after using [https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn](url), Error is same. If you want, I can provide credentials for test.
> > 
> > ![Screenshot (6)](https://private-user-images.githubusercontent.com/28692876/337727575-50b86209-8b99-4996-9b4b-14838db16481.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3OTI0OTYsIm5iZiI6MTcxODc5MjE5NiwicGF0aCI6Ii8yODY5Mjg3Ni8zMzc3Mjc1NzUtNTBiODYyMDktOGI5OS00OTk2LTliNGItMTQ4MzhkYjE2NDgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDEwMTYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjODI4MTA2NDVmNzVmNjE1MDIxYjg1Njk0MTkzMDdmOTQwYzM5NzRmOTcyM2Q2ZmI0MjNhN2M4NWJlOTZiZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.d3FRoTiwiQ_pmYE3zzSjGnq5DarhPkknsN9TkIrxmF4)
> 
> I can look into it, but yes i will need at least a bearer token to test with. you can email me puyodead@proton.me

Done!

**Comment by Puyodead1** [comment]
> > > > are you passing a bearer token? and use `https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn`
> > > 
> > > 
> > > Yes, I am using the token. Even after using [https://www.udemy.com/course/react-the-complete-guide-incl-redux/learn](url), Error is same. If you want, I can provide credentials for test.
> > > ![Screenshot (6)](https://private-user-images.githubusercontent.com/28692876/337727575-50b86209-8b99-4996-9b4b-14838db16481.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3OTI0OTYsIm5iZiI6MTcxODc5MjE5NiwicGF0aCI6Ii8yODY5Mjg3Ni8zMzc3Mjc1NzUtNTBiODYyMDktOGI5OS00OTk2LTliNGItMTQ4MzhkYjE2NDgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDEwMTYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjODI4MTA2NDVmNzVmNjE1MDIxYjg1Njk0MTkzMDdmOTQwYzM5NzRmOTcyM2Q2ZmI0MjNhN2M4NWJlOTZiZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.d3FRoTiwiQ_pmYE3zzSjGnq5DarhPkknsN9TkIrxmF4)
> > 
> > 
> > I can look into it, but yes i will need at least a bearer token to test with. you can email me [puyodead@proton.me](mailto:puyodead@proton.me)
> 
> Done! Please check your email.

I'm looking into this, but this issue only occurs on the master branch, I suggest switching to feat/refactor

**Comment by Puyodead1** [comment]
should be resolved in c7329b1d59edb747a400a800fa49735091acb02c
